pkg/provider/consulcatalog/consul_catalog.go

```diff
@@ -146,7 +146,11 @@ func (p *Provider) Provide(configurationChan chan<- dynamic.Message, pool *safe.
 			// will cause an error condition.
 			if p.ConnectAware {
 				logger.Infof("Waiting for Connect certificate before building first configuration")
-				certInfo = <-p.certChan
+				select {
```

Comment:
While this is not particularly nice, it is needed to properly shutdown when the context is done. Otherwise traefik will complain:
```
panic: Timeout while stopping traefik, killing instance ✝

goroutine 133 [running]:
github.com/traefik/traefik/v2/pkg/server.(*Server).Close.func1(0x3d10520, 0xc0008204e0)
	/home/florian/go/src/github.com/traefik/traefik/pkg/server/server.go:92 +0x114
created by github.com/traefik/traefik/v2/pkg/server.(*Server).Close
	/home/florian/go/src/github.com/traefik/traefik/pkg/server/server.go:87 +0x85
```

Reply:
yep, makes sense.